content/blog/announcing-enum-support/index.md

```diff
@@ -0,0 +1,277 @@
+
+If we look at the [resource docs](https://www.pulumi.com/docs/reference/pkg/aws/s3/bucket/#acl_nodejs), we can see that the `acl` property can only be set to one of a few different values: `private`, `public-read`, `public-read-write`, `aws-exec-read`, `authenticated-read` and `log-delivery-write`. The `acl` property is the perfect candidate for an enum type and is now emitted as one, so you can use the following code instead.
+
+{{< chooser >}}
+{{< choosable language typescript >}}
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import { s3 } from "@pulumi/aws";
+
+const myBucket = new s3.Bucket("myBucket", { acl: s3.CannedAcl.Private });
+```
+
+{{< /choosable >}}
+{{< choosable language python >}}
+
+```python
+import pulumi
+from pulumi_aws import s3
+
+my_bucket = s3.Bucket("myBucket", acl=s3.CannedAcl.PRIVATE)
+```
+
+{{< /choosable >}}
+{{< choosable language csharp >}}
+
+```csharp
+using Pulumi;
+using S3 = Pulumi.Aws.S3;
+
+class MyStack : Stack
+{
+    public MyStack()
+    {
+        var myBucket = new S3.Bucket("myBucket", new S3.BucketArgs
+        {
+            Acl = S3.CannedAcl.Private,
+        });
+    }
+}
+```
+
+{{< /choosable >}}
+{{< choosable language go >}}
+
+```go
+package main
+
+import (
+    "github.com/pulumi/pulumi-aws/sdk/v3/go/aws/s3"
+    "github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+)
+
+func main() {
+    pulumi.Run(func(ctx *pulumi.Context) error {
+        myBucket, err := s3.NewBucket(ctx, "myBucket", &s3.BucketArgs{
+            Acl: s3.CannedAclPrivate,
+        })
+        if err != nil {
+            return err
+        }
+        return nil
+    })
+}
+```
+
+{{< /choosable >}}
+{{< /chooser >}}
+
+## IDE Superpowers
+
+Enum properties provide discoverable and normalized constants that can be used in place of raw strings, allowing you to move faster with the added boosts of IDE tooling like type hints and autocomplete.
+
+By using the provided constants, you can avoid having to refer back to documentation to remember the valid values and save precious moments in the development cycle that would be lost to debugging errors caused by typos.
+
+{{< chooser >}}
+{{< choosable language typescript >}}
+
+![ENUM_TYPESCRIPT](ts-enum.gif)
+
+{{< /choosable >}}
+{{< choosable language python >}}
+
+![ENUM_PYTHON](python-enum.gif)
+
+{{< /choosable >}}
+{{< choosable language csharp >}}
+
+![ENUM_CSHARP](csharp-enum.gif)
+
+{{< /choosable >}}
+{{< choosable language go >}}
+
+![ENUM_GO](go-enum.gif)
+
+{{< /choosable >}}
+{{< /chooser >}}
+
+## Optimized for flexibility
+
+While some properties make sense as "strict" enums (i.e. the input value **must** be one of the enumerated values), there are times when it would be beneficial to have constants for commonly used values without restricting the input to **only** those values. We call this second category "relaxed" enums, and we model them a little differently in our schema.
```

Comment:
```suggestion
While some properties make sense as "strict" enums (i.e,. the input value **must** be one of the enumerated values), there are times when it would be beneficial to have constants for commonly used values without restricting the input to **only** those values. We call this second category "relaxed" enums, and we model them a little differently in our schema.
```

Reply:
@spara this one doesn't look quite right. did you mean `i.e.,`?